homeassistant/components/shelly/switch.py

```diff
@@ -89,6 +96,7 @@ async def async_setup_rpc_entry(
         return
 
     async_add_entities(RpcRelaySwitch(wrapper, id_) for id_ in switch_ids)
+    async_add_entities([ShellyOtaUpdateBetaChannelSwitch(wrapper, config_entry)])
```

Comment:
This needs to be before line 53, so switch is created also for devices without switches ( lights, gas sensor, etc. )

Reply:
Nice catch, I suggest to put it right after we fetch the wrapper
https://github.com/home-assistant/core/blob/b46bba5610d73d61eeb4048657e510cbe785f5a0/homeassistant/components/shelly/switch.py#L49

Same should be for rpc device

Reply:
Exactly where I suggested to place it ;-)

Reply:
done with 5e8c2dd